app/resolve/core/request.py

```diff
@@ -0,0 +1,36 @@
+from requests.models import Response
+from enum import Enum
+import requests
+from requests.auth import HTTPBasicAuth
+from core.settings import USERNAME, PASSWORD
+
+
+class SupportedMethod(Enum):
+    POST = "post"
+    GET = "get"
+    PUT = "put"
+    PATCH = "patch"
+    DELETE = "delete"
+
+
+def request(method: SupportedMethod, url: str, data: dict = None, **kwargs) -> Response:
```

Comment:
```suggestion
def request(method: SupportedMethod, url: str, data: Optional[dict] = None, **kwargs) -> Response:
```
[PEP484](https://peps.python.org/pep-0484/#union-types) prohibits implicit optional.

But I think a wider refactor would be better here. We should really do this `request.py` file as an `task_api_client.py` so we can implement this consistently with the polling as well. 

I'm thinking a class `TaskApiClient` with the `base_url/user/password` as properties.
A `request` method as you've done here
And a `POST` method for the returning results. 
(and soon either a `GET` or polling however that is implemented)

Reply:
I have changed the name and improved the logic of the `request` method, but still had some questions:
- with the current, updated `request` function, I believe it can be used as GET and POST requests, right? Why do we need separate POST and GET methods?
- What is the use of the `TaskApiClient` class? and is the `base_url/user/password` the URL for authentication in Relay?

Reply:
(for the 1st question) did you mean GET method will have the Return <something> and POST method will not?

Reply:
Creating a `TaskApiClient` would encapsulate all the logic for query the task API and keep this maintainable. Like in the C# code this fully exists doing the same thing: https://github.com/Health-Informatics-UoN/hutch-cohort-discovery/blob/main/lib/Hutch.Rackit/TaskApi/TaskApiClient.cs
You could just keep the same request method, just it's not commonly done.